File: crates/ocql-extractor-cpp/tests/fixtures/simple.cpp
// A simple C++ file for testing extraction.

#include <iostream>

int global_var = 42;

struct Point {
    int x;
    int y;
};

class Animal {
public:
    Animal(const char* name) : name_(name) {}
    virtual ~Animal() {}
    virtual void speak() const = 0;
    const char* name() const { return name_; }
private:
    const char* name_;
};

class Dog : public Animal {
public:
    Dog(const char* name) : Animal(name) {}
    void speak() const override {
        std::cout << name() << " says woof!" << std::endl;
    }
};

int factorial(int n) {
    if (n <= 1) return 1;
    return n * factorial(n - 1);
}

int main() {
    Dog d("Rex");
    d.speak();
    Point p = {1, 2};
    int result = factorial(5);
    return 0;
}

File: crates/ocql-e2e-tests/tests/fixtures/Simple.java
// Simple Java class for end-to-end testing.
public class Simple {
    private int value;

    public Simple(int v) {
        this.value = v;
    }

    public int getValue() {
        return value;
    }

    public int add(int x) {
        return value + x;
    }

    public static int helper(int a, int b) {
        return a + b;
    }

    public static void main(String[] args) {
        Simple s = new Simple(42);
        int result = s.add(10);
        int sum = helper(result, 5);
        System.out.println(sum);
    }
}
